src/components/Account/AccountModule.js

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { useWallet } from 'use-wallet'
+import { Button, GU, IconConnect, useViewport } from '@aragon/ui'
+import AccountButton from './AccountButton'
+import AccountPopover from './AccountPopover'
+import ScreenConnected from './ScreenConnected'
+import ScreenConnecting from './ScreenConnecting'
+import ScreenError from './ScreenError'
+import ScreenProviders from './ScreenProviders'
+
+const SCREENS = [
+  { id: 'providers', title: 'Use account from' },
+  { id: 'connecting', title: 'Use account from' },
+  { id: 'connected', title: 'Active account' },
+  { id: 'error', title: 'Connection error' },
+]
+
+function AccountModule() {
+  const [opened, setOpened] = useState(false)
+  const [activatingDelayed, setActivatingDelayed] = useState(null)
+  const buttonRef = useRef()
+  const { below } = useViewport()
+  const compactMode = below('medium')
+  const wallet = useWallet()
+  const { account, connector, error, status } = wallet
+
+  const open = useCallback(() => setOpened(true), [])
+  const toggle = useCallback(() => setOpened((opened) => !opened), [])
+
+  useEffect(() => {
+    let timer
+
+    if (status === 'error') {
+      setActivatingDelayed(null)
+    }
+
+    if (status === 'connecting') {
+      setActivatingDelayed(connector)
+      timer = setTimeout(() => {
+        setActivatingDelayed(null)
+      }, 400)
+    }
+
+    return () => clearTimeout(timer)
+  }, [connector, status])
+
+  const handleResetConnection = useCallback(() => {
+    wallet.reset()
+  }, [wallet])
+
+  const handleActivate = useCallback(
+    (providerId) => wallet.connect(providerId),
+    [wallet]
+  )
+
+  const previousScreenIndex = useRef(-1)
+
+  const { screenIndex, direction } = useMemo(() => {
+    const screenId = status === 'disconnected' ? 'providers' : status
+
+    const screenIndex = SCREENS.findIndex((screen) => screen.id === screenId)
+    const direction = previousScreenIndex.current > screenIndex ? -1 : 1
+
+    previousScreenIndex.current = screenIndex
+
+    return { direction, screenIndex }
+  }, [status])
+
+  const screen = SCREENS[screenIndex]
+  const screenId = screen.id
+
+  const handlePopoverClose = useCallback(() => {
+    if (screenId === 'connecting' || screenId === 'error') {
+      // reject closing the popover
+      return false
+    }
+    setOpened(false)
+  }, [screenId])
+
+  return (
+    <div
+      ref={buttonRef}
+      tabIndex="0"
```

Comment:
Do we need this? it seems to interfere with keyboard tabbing, but I'm curious if it exists to fix some other issue 🤔